src/pages/recipes/events/index.md

```diff
@@ -0,0 +1,139 @@
+
+You can see the sample scripts in order of complexity, starting with very simple scripts and building toward more complex operations.
+>
+*Learn how to create, install, and run a script with [Adobe InDesign UXP Scripting Tutorial](https://developer.adobe.com/indesign/uxp/guides/getting-started/).*
+
+## Understanding Event Scripting
+To respond to an event, you register an eventListener with an object capable of receiving the event. When the specified event reaches the object, the eventListener executes the script function defined in its handler function (which can be either a script function or a reference to a script file on disk).
+>
+*Here is the [list of available events](https://developer.adobe.com/indesign/dom/api/e/Event/).*
+
+## Working with Event Listeners
+
+**Adding an eventListener**
+
+When you create an eventListener, you specify the event type and the event handler (as a function or file reference). The following script fragment shows how to add an eventListener for a specific event (for the complete script, see AddEventListener).
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+let myEventListener = app.addEventListener("afterNew", myDisplayEventType);
+```
+The preceding script fragment refers to the following function:
+
+```js
+function myDisplayEventType(myEvent){
+    alert("This event is the " + myEvent.eventType + " event.");
+}
+```
+
+**Removing an eventListener**
+
+To remove the eventListener created by the preceding script, run the following script:
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+let myResult = app.removeEventListener("afterNew", myDisplayEventType);
+```
+When an eventListener responds to an event, the event may still be processed by other eventListeners that might be monitoring the event (depending on the propagation of the event). For example, the afterOpen event can be observed by eventListeners associated with both the application and the document.
+
+eventListeners do not persist beyond the current InDesign session. To make an eventListener available in every InDesign session, add the script to the startup scripts folder. 
+
+An event can trigger multiple eventListeners as it propagates through the scripting object model. The following sample script demonstrates an event triggering eventListeners registered to different objects:
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+let myApplicationEventListener = app.eventListeners.add("beforeImport", myEventInfo);
+let myDocumentEventListener = app.documents.item(0).eventListeners.add("beforeImport", myEventInfo);
+function myEventInfo(myEvent){
+    let myString = "Current Target: " + myEvent.currentTarget.name;
+    alert(myString);
```

Comment:
alert not declared. Use console.log instead.

Reply:
Done